sstables/generation_type.hh

```diff
@@ -130,6 +130,11 @@ public:
         _last_generation += seastar::smp::count;
         return generation_type(_last_generation);
     }
+    /// returns a hint indicating if an sstable belongs to a shard is determined by
+    /// overlapping its partition-ranges with the shard's owned ranges.
```

Comment:
This feels a little bit mashed together, feels like it needs a comma: "returns a hint indicating if an sstables belongs to a shard. the definitive way to determine that is overlapping...".